src/Apps/Gene/Components/GeneArtworkFilter.tsx

```diff
@@ -87,10 +93,81 @@ export const GeneArtworkFilterRefetchContainer = createRefetchContainer(
     `,
   },
   graphql`
-    query GeneArtworkFilterQuery($slug: String!, $input: FilterArtworksInput) {
+    query GeneArtworkFilterRefetchQuery(
+      $slug: String!
+      $input: FilterArtworksInput
+    ) {
       gene(id: $slug) {
         ...GeneArtworkFilter_gene @arguments(input: $input)
       }
     }
   `
 )
+
+interface GeneArtworkFilterQueryRendererProps {}
+
+export const GeneArtworkFilterQueryRenderer: React.FC<GeneArtworkFilterQueryRendererProps> = rest => {
+  const { relayEnvironment } = useSystemContext()
+  const { match } = useRouter()
+
+  return (
+    <SystemQueryRenderer<GeneArtworkFilterQuery>
```

Comment:
I think you might also need that `<StaticContainer>` bit from my pr so that things dont rerender if changing URL when clicking away. Can you verify that you don't get big full screen flashes when clicking on an artwork?

Reply:
So I am still seeing a bit of a flash (I think) when clicking an artwork in the grid, but adding the `StaticContainer` wrapper around the query renderer:

``` ts
  <StaticContainer shouldUpdate={!!match.elements}>
    <GeneArtworkFilterQueryRenderer />
  </StaticContainer>
```

didn't seem to do much.

Not 100% sure.

Reply:
Actually currently on staging the same kind of 'flash' I'm seeing locally, I see there too.

Try https://staging.artsy.net/gene/animals logged out for instance.


https://github.com/user-attachments/assets/14e8ec8d-f3f4-4dc4-83be-15d5b21a90d6

Reply:
Given that, I don't see anything _worse_ in this PR, nor do I see anything different when using `StaticContainer` wrapping (although admittedly I don't quite understand that).

So I think I'd merge this as-is and we could follow up.

Reply:
ohhh that's something else -- just the bundle loading. So yah. Merge away. I was talking about the artwork grid skeleton preloading suddenly appearing. 

But now that i think about it, with prefetching on, it is very likely we'll ever see that or get in that state, since the data is being loaded in the background.

Reply:
The pattern is, just fyi:

- Artwork grid lazy loads after showing skeleton
- User clicks artwork, which updates the URL
- The url change triggers a rerender in the artwork grid while the data is fetching and the top loading bar appears. 
- Rerender happens because we're using `useRouter()` and looking at the match.location.query or match.params. 
- So we need to use the static container to check against rerenders when the loading bar is visible (via `!match.elements` -- no elements things are fetching)

Reply:
Ah I see, the artwork grid placeholder itself showing up! I do not see that behavior w/ or w/o the static container stuff - but once it lands in staging/prod can double-check. Sometimes local can be harder (or easier) to notice that stuff.